airflow/providers/asana/hooks/asana.py

```diff
@@ -0,0 +1,271 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Connect to Asana."""
+import logging
+from typing import Any, Dict
+
+from cached_property import cached_property
+
+from airflow.hooks.base import BaseHook
+from asana import Client
+from asana.error import NotFoundError
+
+log = logging.getLogger(__name__)
+
+
+class AsanaHook(BaseHook):
+    """Wrapper around Asana Python client library."""
+
+    conn_name_attr = "asana_conn_id"
+    default_conn_name = "asana_default"
+    conn_type = "asana"
+    hook_name = "Asana"
+
+    def __init__(self, conn_id: str = default_conn_name, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.connection = self.get_connection(conn_id)
+        extras = self.connection.extra_dejson
+        self.workspace = extras.get("extra__asana__workspace") or None
+        self.project = extras.get("extra__asana__project") or None
+
+    def get_conn(self) -> Client:
+        return self.client
+
+    @staticmethod
+    def get_connection_form_widgets() -> Dict[str, Any]:
+        """Returns connection widgets to add to connection form"""
+        from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import StringField
+
+        return {
+            "extra__asana__workspace": StringField(lazy_gettext("Workspace"), widget=BS3TextFieldWidget()),
+            "extra__asana__project": StringField(lazy_gettext("Project"), widget=BS3TextFieldWidget()),
+        }
+
+    @staticmethod
+    def get_ui_field_behaviour() -> Dict:
+        """Returns custom field behaviour"""
+        return {
+            "hidden_fields": ["port", "host", "login", "schema"],
+            "relabeling": {},
+            "placeholders": {
+                "password": "Asana personal access token",
+                "extra__asana__workspace": "Asana workspace gid",
+                "extra__asana__project": "Asana project gid",
+            },
+        }
+
+    @cached_property
+    def client(self) -> Client:
+        """Instantiates python-asana Client"""
+        if not self.connection.password:
+            raise ValueError(
+                "Asana connection password must contain a personal access token: "
+                "https://developers.asana.com/docs/personal-access-token"
+            )
+
+        return Client.access_token(self.connection.password)
+
+    def create_task(self, task_name: str, params: dict) -> dict:
+        """
+        Creates an Asana task.
+        :param task_name: Name of the new task
```

Comment:
```suggestion
        Creates an Asana task.
        
        :param task_name: Name of the new task
```
I think we will need spaces here (in other places) to make doc styl check happy

Reply:
You were right, thanks!